apps/api_web/lib/api_web/plugs/rate_limiter.ex

```diff
@@ -11,15 +11,32 @@ defmodule ApiWeb.Plugs.RateLimiter do
 
   def call(%{assigns: assigns, request_path: request_path} = conn, _) do
     case ApiWeb.RateLimiter.log_request(assigns.api_user, request_path) do
-      :ok ->
+      %{status: :ok, limit_data: %{limit: limit, remaining: remaining, reset_ms: reset_ms}} ->
         conn
+        |> put_rate_limit_headers(limit, remaining, reset_ms)
 
-      {:error, :rate_limited} ->
+      %{status: :ok} ->
         conn
+
+      %{
+        status: {:error, :rate_limited},
+        limit_data: %{limit: limit, remaining: remaining, reset_ms: reset_ms}
+      } ->
+        conn
+        |> put_rate_limit_headers(limit, remaining, reset_ms)
         |> put_status(429)
         |> put_view(ApiWeb.ErrorView)
         |> render("429.json-api", [])
         |> halt()
     end
   end
+
+  defp put_rate_limit_headers(conn, limit, remaining, reset_ms) do
+    reset_seconds = div(reset_ms, 1_000)
+
+    conn
+    |> put_resp_header("x-ratelimit-limit", "#{limit}")
+    |> put_resp_header("x-ratelimit-remaining", "#{remaining}")
+    |> put_resp_header("x-ratelimit-reset", "#{reset_seconds}")
```

Comment:
I don't think there's a test for this?

Reply:
Also, https://opensource.zalando.com/restful-api-guidelines/#153 says reset is a relative number of seconds: do we want to use that instead?

Reply:
I don't have a strong sense of whether developers would prefer to be told a relative number of seconds or a timestamp for this. (Do you?)

I went with timestamp over relative because that's what Twitter, Github, Vimeo and Twitch all do; I didn't come across any examples of APIs from well-known companies that include a response header similar to `X-RateLimit-Reset` where the value is relative.

Reply:
I think the rationale for having a relative timestamp is that it avoids the issue where different servers have a different time. However, I don't have a strong preference and other companies have standardized on an epoch time, so let's keep it this way.